csrc/scheduler/pointwise_utils.cpp

```diff
@@ -142,6 +144,104 @@ bool DomainMap::areAllInputIdsMappedTo(TensorView* input_tv, TensorView* tv)
   return in_concrete_ids.empty();
 }
 
+// Note: ideally we would want to chck that reference_tv contains (not
+// necessarily maps) all iter domain in output_tv, so that transformation
+// applied on reference_tv can be propagated to output_tv. But we don't have
+// an easy way to check that.
+// Instead of that, this function checks that all source iter domains involved
+// in transformation on output_tv is covered by reference_tv. We do so by
+// traverse all disjoint set producers on both tvs and filter them with
+// `ca_map_.uniqueExactDefinitions(id).empty()`.
+//
+// ------
+//
+// e.g 0.
+//   T34 [i0, i1]
+//   T185 [i0, b2, i1]     = broadcast(T34)
+//   T192 [i0, b3(ex), i1] = expand(T185)
+//   T198 [i0, b3(ex)*i1]  = reshape(T192)
+//   output(T34)
+//   output(T198)
+//
+// if we consider taking T34 as reference_tv. T198 is the output_tv. We can't
+// replay T34's transform of merging all the dimensions to T198, since b3(ex)*i1
+// can't be reversed. The check in this function would give us T34 with source
+// i0, i1; where T198 would have source i0, b3, i1, where b3 isn't contained in
+// T34. Hence we'll reject this referenc_tv.
```

Comment:
```suggestion
// T34. Hence we'll reject this reference_tv.
```
In this case we could accept `T185`?